packages/x-data-grid/src/components/columnSelection/GridHeaderCheckbox.tsx

```diff
@@ -36,7 +36,9 @@ const GridHeaderCheckbox = React.forwardRef<HTMLButtonElement, GridColumnHeaderP
     const ownerState = { classes: rootProps.classes };
     const classes = useUtilityClasses(ownerState);
     const tabIndexState = useGridSelector(apiRef, gridTabIndexColumnHeaderSelector);
-    const selection = useGridSelector(apiRef, gridRowSelectionStateSelector);
+    const selection = useGridSelector(apiRef, () =>
+      gridRowSelectionStateSelector(apiRef.current.state),
+    );
```

Comment:
How is this different? As far as I recall, this should be selecting the same value.

Reply:
Added it in error. Removed.